flowetl/flowetl/flowetl/util.py

```diff
@@ -390,3 +390,96 @@ def create_dag(
         ]
     globals()[dag_id] = dag
     return dag
+
+
+def create_staging_dag(start_date: datetime, event_types: List[str], end_date=None):
+    """
+    Returns a DAG for moving data from event-CSVs a given day to the reduced sightings format.
+    See individual operators for details.
+
+    Parameters
+    ----------
+    start_date: datetime
+        Date to start mounting CSVs from.
+    event_types: List[str]
+        Event types available for staging. Can be call,sms,location,mds,topup.
+    end_date: datetime
+        Date to stop mounting CSVs on. Leave blank for present day.
+
+    Returns
+    -------
+    dag: DAG
+        The staging DAG.
+    """
+    from airflow import DAG
+    import os
+
+    template_folder = (
+        "/"
+        + os.getenv("SOURCE_TREE")
+        + "/flowetl/flowetl/flowetl/operators/staging/sql"
+    )
+
+    with DAG(
+        # How do we want to trigger this? Ask James./Jono
+        dag_id="load_records_from_staging_dag",
+        start_date=start_date,  # Put this back before pr
+        end_date=end_date,
+        # Defined in the docker-config.yml
+        default_args={"owner": "airflow", "postgres_conn_id": "flowdb"},
+        params={
+            "flowdb_csv_dir": os.getenv("FLOWDB_CSV_DIR"),
+        },
+        template_searchpath=template_folder,
+        is_paused_upon_creation=True,
+    ) as dag:
+        from flowetl.operators.staging.create_and_fill_day_sightings_table import (
+            CreateAndFillDaySightingsTable,
+        )
+        from flowetl.operators.staging.create_sightings_table import (
+            CreateSightingsTable,
+        )
+        from flowetl.operators.staging.default_location_mapping import (
+            DefaultLocationMapping,
+        )
+        from flowetl.operators.staging.append_sightings_to_main_table import (
+            AppendSightingsToMainTable,
+        )
+        from flowetl.operators.staging.apply_mapping_to_staged_events import (
+            ApplyMappingToStagedEvents,
+        )
+        from flowetl.operators.staging.create_and_fill_staging_table import (
+            CreateAndFillStagingTable,
+        )
+        from flowetl.operators.staging.cleanup_staging_table import CleanupStagingTable
+        from flowetl.operators.staging.mount_event_operator_factory import (
+            create_mount_event_operator,
+        )
+
+        event_mount_operator_classes = [
+            create_mount_event_operator(event_type=event_type)
+            for event_type in event_types
+        ]
+        event_mount_instantiations = [op() for op in event_mount_operator_classes]
```

Comment:
As mentioned, I think the dag failing if one event type is missing on one day is an issue. I've thought it over, and my feeling is that while the monitoring should of course report this, creating a bogus file to enable this to pass or whatever isn't really tenable and will cause pain in future, as will attaching a bunch of empty partitions.